aspnet/fundamentals/routing.rst

```diff
@@ -1,10 +1,272 @@
-.. include:: /../common/stub-topic.txt
+Routing
+=======
+By `Steve Smith`_
 
-.. _routing-index:
+Routing middleware is used to map requests to route handlers. Routes are configured when the application starts up, and can define values that will be passed as parameters to route handlers. Routing functionality is also responsible generating links that correspond to routes in ASP.NET apps.
```

Comment:
@ardalis Excellent - Yes, I think we **Need** a best practices for routing. @rynowak @shirhatti @danroth27 @Eilon  can you come up with some best practices?  When I added a simple intro to routing to my “Intro to MVC”, I had lots of customers think they needed to use that feature and ended up shooting themselves in the foot. Here's my (probably bogus) routing best practices:

- Many (most?) apps won't need custom routes. The default route will work in most cases
- Don't use them unless you really need to and you understand all the repercussions
- Routes can be tricky to test and debug
- You can't secure your controller/methods with routes, don't even try

-